backend/packages/Upgrade/src/api/controllers/validators/MarkExperimentValidator.v4.ts

```diff
@@ -0,0 +1,19 @@
+import { IsNotEmpty, IsDefined } from 'class-validator';
+import { MARKED_DECISION_POINT_STATUS } from 'upgrade_types';
+
+export class MarkExperimentValidatorv4 {
+  @IsNotEmpty()
+  @IsDefined()
+  public userId: string;
+
+  public data: {
+    site: string;
+    target: string | undefined;
+    assignedCondition: { conditionCode: string; experimentId: string };
+    assignedFactor: object | undefined;
```

Comment:
as noted in another comment, `object` is not a great type, can we give this a real type for `assignedFactor`?